docs/source/essentials/subscriptions.mdx

```diff
@@ -19,18 +25,71 @@ subscription NumberIncremented {
 
 Unlike with queries and mutations, a subscription operation can include only _one_ field of the `Subscription` type. To subscribe to multiple fields, you create multiple subscription operations.
 
-## Setting the WebSocket URL
+## Configuring WebSockets subscriptions
 
-Because subscriptions usually use WebSocket instead of HTTP, you might have to customize the URL used:
+To use subscriptions over WebSockets, use `WebSocketNetworkTransport` 
 
 ```kotlin
 val apolloClient = ApolloClient.Builder()
-  .webSocketServerUrl("https://apollo-fullstack-tutorial.herokuapp.com/graphql")
-  .build()
+    .subscriptionNetworkTransport(
+        WebSocketNetworkTransport.Builder()
+            .serverUrl("https://apollo-fullstack-tutorial.herokuapp.com/graphql")
+            .build()
+    )
+    .build()
 ```
 
 > **Note:** Apollo Kotlin supports both `https://` (or `http://`) and `wss://` (or `ws://`) protocols. Internally, `wss://` is renamed to `https://` and which one you use does not matter.
 
+
+### Customizing your WebSocket protocol
+
+The GraphQL spec does not specify a particular protocol to use for subscription operations. By default, Apollo Kotlin uses [subscriptions-transport-ws](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md) for backward compatibility purposes, but it supports all the following:
+
+* [subscriptions-transport-ws](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md) (⚠️ not actively maintained!)
+* [graphql-ws](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md)
+* [appsync](https://docs.aws.amazon.com/appsync/latest/devguide/real-time-websocket-client.html) (also uses `graphql-ws` as [Sec-WebSocket-Protocol](https://datatracker.ietf.org/doc/html/rfc6455#section-1.2))
+
+To customize your protocol, use the `WsProtocol` interface. Apollo Kotlin comes with built-in support for the protocols above:
+
+* subscriptions-transport-ws: `SubscriptionWsProtocol` (default)
+* graphql-ws: `GraphQLWsProtocol`
+* appsync: `AppSyncWsProtocol`
+
+For example, you can configure a `graphql-ws` transport like so:
+
+```kotlin
+val apolloClient = ApolloClient.Builder()
+    .subscriptionNetworkTransport(
+        WebSocketNetworkTransport.Builder()
+            .protocol(GraphQLWsProtocol.Factory()) // highlight-line
+            .serverUrl("https://apollo-fullstack-tutorial.herokuapp.com/graphql")
+            .build()
+    )
+    .build()
+```
+
+### Authentication
+
+Please refer to [this section](../advanced/authentication#authenticating-your-websockets) about authentication with WebSocket.
+
+
+## Configuring HTTP subscriptions
+
+To use subscriptions over WebSockets, use `HttpNetworkTransport`
```

Comment:
Is it correct to say "over WebSockets" here?

Reply:
Completely not! Thanks